content/en/blog/2024/otel-collector-container-log-parser/index.md

```diff
@@ -0,0 +1,196 @@
+specification or maintenance of the implementation details. With the proposal
+and implementation of the new
+[container parser](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31959),
+all these implementation details are encapsulated and handled within the
+parser's implementation. Adding to this the ability to cover the implementation
+with unit tests and various fail-over logic indicates a significant improvement
+in container log parsing.
+
+## How container logs look like
+
+First of all let's quickly recall the different container log formats that can
+be met out there:
+
+- Docker container logs:
+
+  `{"log":"INFO: This is a docker log line","stream":"stdout","time":"2024-03-30T08:31:20.545192187Z"}`
+
+- cri-o logs:
+
+  `2024-04-13T07:59:37.505201169-05:00 stdout F This is a cri-o log line!`
+
+- Containerd logs:
+
+  `2024-04-22T10:27:25.813799277Z stdout F This is an awesome containerd log line!`
+
+We can notice that cri-o and containerd log formats are quite similar (both
+follow the CRI logging format) but with a small difference in the timestamp
+format.
+
+To properly handle these 3 different formats you need 3 different routes of
+[stanza](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/stanza)
+operators as we can see in the
+[container parser operator issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31959).
+
+In addition, the CRI format can provide partial logs which you would like to
+combine them into one at first place:
+
+```text
+2024-04-06T00:17:10.113242941Z stdout P This is a very very long line th
+2024-04-06T00:17:10.113242941Z stdout P at is really really long and spa
+2024-04-06T00:17:10.113242941Z stdout F ns across multiple log entries
+```
+
+Ideally you would like our parser to be capable of automatically detecting the
+format at runtime and properly parse the log lines. We will see later that the
+container parser will do that for us.
+
+## Attribute handling
+
+Container log files follow a specific naming pattern from which you can extract
+useful metadata information during parsing. For example, from
+`/var/log/pods/kube-system_kube-scheduler-kind-control-plane_49cc7c1fd3702c40b2686ea7486091d3/kube-scheduler/1.log`,
+you can extract the namespace, the name and UID of the pod, and the name of the
+container.
+
+After extracting this metadata, you need to store it properly using the
+appropriate attributes following the
+[Semantic Conventions](/docs/specs/semconv/resource/k8s/). This handling can
+also be encapsulated within the parser's implementation, eliminating the need
+for users to define it manually.
+
+## Using the new container parser
+
+With all these in mind, the container parser can be configured like this:
+
+```yaml
+receivers:
+  filelog:
+    include_file_path: true
+    include:
+      - /var/log/pods/*/*/*.log
+    operators:
+      - id: container-parser
+        type: container
+```
```

Comment:
How about showing the comparison to the config previously (with the container parser) required?
Or at least mention that the config now is 8 lines long vs. ~80 lines to achieve the same result.

Reply:
Good point! Added at 44778bd, thank's.